lib/tezos/lib/assets/user-data/node.sh

```diff
@@ -0,0 +1,215 @@
+
+aws configure set default.s3.max_concurrent_requests 50
+aws configure set default.s3.multipart_chunksize 256MB
+
+echo 'Installing SSM Agent'
+yum install -y $SSM_AGENT_BINARY_URI
+
+echo "Installing s5cmd"
+cd /opt
+wget -q $S5CMD_URI -O s5cmd.tar.gz
+tar -xf s5cmd.tar.gz
+chmod +x s5cmd
+mv s5cmd /usr/bin
+s5cmd version
+
+if [[ "$STACK_ID" != "none" ]]; then
+  echo "Install CloudFormation helper scripts"
+  mkdir -p /opt/aws/
+  pip3 install https://s3.amazonaws.com/cloudformation-examples/aws-cfn-bootstrap-py3-latest.tar.gz
+  sudo ln -s /usr/local/init/ubuntu/cfn-hup /etc/init.d/cfn-hup
+
+  echo "Configuring CloudFormation helper scripts"
+  mkdir -p /etc/cfn/
+  mv /opt/cfn-hup/cfn-hup.conf /etc/cfn/cfn-hup.conf
+  sed -i "s;__AWS_STACK_ID__;\"$STACK_ID\";g" /etc/cfn/cfn-hup.conf
+  sed -i "s;__AWS_REGION__;\"$AWS_REGION\";g" /etc/cfn/cfn-hup.conf
+
+  mkdir -p /etc/cfn/hooks.d/
+  mv /opt/cfn-hup/cfn-auto-reloader.conf /etc/cfn/hooks.d/cfn-auto-reloader.conf
+  sed -i "s;__AWS_STACK_NAME__;\"$STACK_NAME\";g" /etc/cfn/hooks.d/cfn-auto-reloader.conf
+  sed -i "s;__AWS_REGION__;\"$AWS_REGION\";g" /etc/cfn/hooks.d/cfn-auto-reloader.conf
+
+  echo "Starting CloudFormation helper scripts as a service"
+  mv /opt/cfn-hup/cfn-hup.service  /etc/systemd/system/cfn-hup.service
+
+  systemctl daemon-reload
+  systemctl enable --now cfn-hup
+  systemctl start cfn-hup.service
+
+fi
+
+echo "Install Octez-node and its dependencies"
+
+if [ "$arch" == "x86_64" ]; then
+    curl -o octez-binaries.tar.gz https://gitlab.com/tezos/tezos/-/package_files/133747462/download
+    tar xf octez-binaries.tar.gz
+    mv ./octez-arm64/* /usr/local/bin/
+    else
+    curl -o octez-binaries.tar.gz https://gitlab.com/tezos/tezos/-/package_files/133748628/download
+    tar xf octez-binaries.tar.gz
+    mv ./octez-arm64/* /usr/local/bin/
+fi
+
+
+
+find /usr/local/bin/ -name "octez-*" -exec chmod +x {} \;
+groupadd tezos
+adduser -g tezos tezos
+#mkdir -p /var/tezos/node
+#chown -R tezos:tezos /var/tezos
+
+echo "Changing user to tezos"
+
+
+echo "Installing zcash dependency"
+curl -o  /tmp/fetch-params.sh https://raw.githubusercontent.com/zcash/zcash/713fc761dd9cf4c9087c37b078bdeab98697bad2/zcutil/fetch-params.sh
+chmod +x /tmp/fetch-params.sh
+su tezos -c "/tmp/fetch-params.sh"
+
+echo "Configuring node"
+su tezos -c "octez-node config init  --network=$TZ_NETWORK  --history-mode=$TZ_HISTORY_MODE  --net-addr='[::]:9732' --rpc-addr='[::]:8732' --allow-all-rpc [::]:8732"
+
+
+# download snapshot if network is mainnet
+if [ "$INSTANCE_TYPE" == "SNAPSHOT"  ] || [ "$INSTANCE_TYPE" == "SINGLE" ]; then
+  if [ "$TZ_NETWORK" == "mainnet"  ] && [ "$TZ_DOWNLOAD_SNAPSHOT" == "true" ]; then
+    echo "Downloading Tezos snapshot and importing"
+    chmod +x /opt/download-snapshot.sh
+    su tezos -c "/opt/download-snapshot.sh"
+  fi
+fi
+
+
+if [[ "$INSTANCE_TYPE" == "HA" ]]; then
+  su tezos -c "aws s3 sync s3://$S3_SYNC_BUCKET/node ~/.tezos-node/node"
```

Comment:
You can use a faster `s5cmd` tool than the original `aws s3 sync`. We installed it earlier on lines 68-72.

Reply:
✅

Reply:
✅